### ISSUE TITLE
Query language: Support "has no key like" 

### ISSUE THREAD

**Issue by westnordost** [drolbr/Overpass-API#589]
What already exists is `[!key]` and `[~kexRegex]`. What is missing is `[!~keyRegex]` or similar to check if an element has **no** key that matches with the regex tagged.

Example use cases:
- Check if a recycling container is not tagged with any recycleable it accepts: `[!~'^recycling:.*$']`. 
- Check if the road has no cycleway tagged: `!~'^cycleway(:left|:right|:both)?$'`
- There are other such schemes as the recycling scheme, at least I think for services provided at car shops, boat shops, motorcycle shops

**Comment by mmd-osm** [comment]
This wasn't implemented for performance reasons, as it would be prohibitively expensive. I would have to look up a reference to a previous discussion on this topic if needed.

**Comment by westnordost** [comment]
Ok, I understand

**Comment by westnordost** [comment]
Ah wait, but perhaps as an evaluator?

`if: hasTagLike("^recycling:.*$")`

**Comment by mmd-osm** [comment]
Evaluators don't support any regular expressions at this time...

**Comment by westnordost** [comment]
Ok. But if they do in the future, that might be a thing to consider.

**Comment by westnordost** [comment]
I further [worked on the "element filter syntax"](
https://github.com/westnordost/StreetComplete/issues/1998#issuecomment-670577017) used by StreetComplete by the way and found that for everything I need to conveniently support these extensions exists in OQL 👍

**Comment by mmd-osm** [comment]
Found this commit: the XML query language rejects regular expressions w/ negation: https://github.com/drolbr/Overpass-API/commit/7209dd399b8abfadbc8f413cb914ac0c0be81d35

**Comment by mmd-osm** [comment]
Just wanted to mention that you would typically use a difference statement in this case:

```
(nwr[recycling]->.a; - nwr.a[~"^recycling:."~"."];);
out meta center;
```

**Comment by westnordost** [comment]
Yes, this is how I currently do it

**Comment by mmd-osm** [comment]
Yeah, I know :) this was meant as hint to people who found this issue via some web search.

**Comment by drolbr** [comment]
I do not see any inevitable performance problem here. When other conditionals restrict the result set or the expression is sufficiently broadly (think of `way[!~"."~"."]`, finding all ways without any tag) then it is probably fine.

It is just not a high priority.

**Comment by mmd-osm** [comment]
Prototype implementation is here for testing: https://overpass-turbo.eu/s/1ko6
